spec/casclient/client_spec.rb

```diff
@@ -90,4 +90,12 @@
       end
     end
   end
+
+  describe 'logout_url' do
+    it 'should tolerate oddly formed URIs' do
+      destination_url = 'https://foo.com/path/to/page?&sort=date_received.DESC'
+      
+      client.logout_url(destination_url)
```

Comment:
Ugh, it's been three years and the underlying bug is still present - https://github.com/rubycas/rubycas-client/blob/master/lib/casclient/client.rb#L286